src/pages/SmartWeaveExplorer/Contract/Contract.vue

```diff
@@ -580,22 +586,31 @@ export default {
         this.getInteractions(this.currentPage, this.selected);
       }
     },
+
     async getContract() {
-      axios
-        .get(`${this.gatewayUrl}/gateway/contract?txId=${this.contractId}`)
-        .then((fetchedContract) => {
-          this.owner = fetchedContract.data.owner;
-          this.pst_ticker = fetchedContract.data.pstTicker;
-          this.pst_name = fetchedContract.data.pstName;
-          this.wasmLang = fetchedContract.data.srcWasmLang;
-          this.initState = fetchedContract.data.initState;
-          this.loadedContract = true;
-          this.sourceTxId = fetchedContract.data.srcTxId;
-        })
-        .catch((e) => {
-          this.correct = false;
-        });
+      const response = await fetch(`${this.gatewayUrl}/gateway/contract?txId=${this.contractId}`);
+      if (!response.ok) {
+        const error = `An error has occured: ${response.status}`;
+        throw new Error(error);
+      }
```

Comment:
This one is pretty bad. Error is not caught anyhow. You would need to switch `this.correct` flag to `false`.

Reply:
fixed